packages/@aws-cdk/aws-ec2/lib/vpc.ts

```diff
@@ -456,7 +456,7 @@ export class VpcNetwork extends VpcNetworkBase {
     if (placement) {
       const subnets = this.subnets(placement);
       for (const sub of subnets) {
-        if (!this.isPublicSubnet(sub)) {
+        if (this.publicSubnets.indexOf(sub) === -1) {
```

Comment:
Why not use the function you created?

Reply:
Because it feels silly to convert all subnet objects to IDs just to match on IDs.

I just realized it's a miracle this works at all, btw, because the subnet IDs are Tokens of course :).